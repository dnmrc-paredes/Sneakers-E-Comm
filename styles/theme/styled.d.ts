import 'styled-components'

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      orange: string
      pale_orange: string
      light_gray: string
    }
    fonts: {
      nunito: string
      kumbh: string
    }
  }
}
