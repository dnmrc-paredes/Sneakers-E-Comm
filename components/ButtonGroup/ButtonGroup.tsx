import React from 'react'
import Image from 'next/image'
import styled from 'styled-components'
import { useCart } from '../../contexts/CartContext'

const S = {
  Buttons: styled.div`
    flex: 1;
    display: flex;
    border-radius: 10px;
    background-color: hsl(223, 64%, 98%);
    width: 100%;
  `,
  Minus: styled.button`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    border: none;
    outline: none;
    background-color: transparent;
    cursor: pointer;
  `,
  Count: styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    font-family: ${({ theme }) => theme.fonts.kumbh};
    font-weight: 800;
  `,
  Plus: styled.button`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    border: none;
    outline: none;
    background-color: transparent;
    cursor: pointer;
    padding: 20px 0;
  `
}

export const ButtonGroup = () => {
  const { count, setCount } = useCart()

  return (
    <S.Buttons>
      <S.Minus
        data-testid='minusBtn'
        onClick={() => setCount(prev => (prev === 0 ? 0 : prev - 1))}
      >
        <Image src='/icon-minus.svg' width={15} height={5} alt='Minus' />
      </S.Minus>
      <S.Count data-testid='count'>{count}</S.Count>
      <S.Plus data-testid='plusBtn' onClick={() => setCount(prev => prev + 1)}>
        <Image src='/icon-plus.svg' width={15} height={15} alt='Plus' />
      </S.Plus>
    </S.Buttons>
  )
}
