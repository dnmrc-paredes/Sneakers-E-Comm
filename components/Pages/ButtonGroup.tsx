import React, { Dispatch, SetStateAction } from 'react'
import Image from 'next/image'
import * as Root from './index'
import PlusIcon from '../../public/icon-plus.svg'
import MinusIcon from '../../public/icon-minus.svg'

type ButtonGroupProps = {
  count: number
  setCount: Dispatch<SetStateAction<number>>
}

export const ButtonGroup = ({ count, setCount }: ButtonGroupProps) => {
  return (
    <Root.Buttons>
      <Root.Minus onClick={() => setCount((prev) => prev === 0 ? prev = 0 : prev-=1)}>
        <Image src={MinusIcon} alt='Minus' />
      </Root.Minus>
      <Root.Count>{count}</Root.Count>
      <Root.Plus onClick={() => setCount((prev) => (prev += 1))}>
        <Image src={PlusIcon} alt='Minus' />
      </Root.Plus>
    </Root.Buttons>
  )
}
