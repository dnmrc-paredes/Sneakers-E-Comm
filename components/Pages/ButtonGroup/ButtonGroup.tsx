import React, { Dispatch, SetStateAction } from 'react'
import Image from 'next/image'
import * as Root from '../index'

type ButtonGroupProps = {
  count: number
  setCount: Dispatch<SetStateAction<number>>
}

export const ButtonGroup = ({ count, setCount }: ButtonGroupProps) => {
  return (
    <Root.Buttons>
      <Root.Minus
        data-testid='minusBtn'
        onClick={() =>
          setCount(prev => (prev === 0 ? (prev = 0) : (prev -= 1)))
        }
      >
        <Image src='/icon-minus.svg' width={15} height={5} alt='Minus' />
      </Root.Minus>
      <Root.Count>{count}</Root.Count>
      <Root.Plus
        data-testid='plusBtn'
        onClick={() => setCount(prev => (prev += 1))}
      >
        <Image src='/icon-plus.svg' width={15} height={15} alt='Plus' />
      </Root.Plus>
    </Root.Buttons>
  )
}
