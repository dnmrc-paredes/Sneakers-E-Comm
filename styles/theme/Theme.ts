export const Theme = {
  colors: {
    orange: 'hsl(26, 100%, 55%)',
    pale_orange: 'hsl(25, 100%, 94%)',
    light_gray: 'hsl(220, 14%, 75%)'
  },
  fonts: {
    nunito: 'Nunito Sans',
    kumbh: 'Kumbh Sans'
  }
}
