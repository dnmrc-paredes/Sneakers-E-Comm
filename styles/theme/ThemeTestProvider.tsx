import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Theme } from './Theme'

export const ThemeTestProvider = ({
  children
}: {
  children: React.ReactNode
}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>
}
