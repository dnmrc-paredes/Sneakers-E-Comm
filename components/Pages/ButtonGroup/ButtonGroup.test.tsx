import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { ButtonGroup } from './ButtonGroup'
import { ThemeTestProvider } from '../../../styles/theme/ThemeTestProvider'

describe('ButtonGroup component', () => {
  const count = 0
  const setCount = jest.fn()

  it('Render the ButtonGroup', () => {
    const { getByTestId } = render(
      <ThemeTestProvider>
        <ButtonGroup count={count} setCount={setCount} />
      </ThemeTestProvider>
    )
    expect(getByTestId('count')).toBeInTheDocument()
  })

  it('Match the snapshot', () => {
    const tree = renderer.create(
      <ThemeTestProvider>
        <ButtonGroup count={count} setCount={setCount} />
      </ThemeTestProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
