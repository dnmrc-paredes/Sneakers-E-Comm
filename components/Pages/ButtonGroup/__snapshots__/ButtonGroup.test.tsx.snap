// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup component Match the snapshot 1`] = `
<div
  className="Pages__Buttons-sc-qwt8hg-13 bZjPAx"
>
  <button
    className="Pages__Minus-sc-qwt8hg-15 fIZRuM"
    data-testid="minusBtn"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUiIGhlaWdodD0iNSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="Minus"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="Minus"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=%2Ficon-minus.svg&w=32&q=75"
          srcSet="/_next/image?url=%2Ficon-minus.svg&w=16&q=75 1x, /_next/image?url=%2Ficon-minus.svg&w=32&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </button>
  <div
    className="Pages__Count-sc-qwt8hg-14 imGiZA"
    data-testid="count"
  >
    0
  </div>
  <button
    className="Pages__Plus-sc-qwt8hg-16 cirGoH"
    data-testid="plusBtn"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUiIGhlaWdodD0iMTUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="Plus"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="Plus"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=%2Ficon-plus.svg&w=32&q=75"
          srcSet="/_next/image?url=%2Ficon-plus.svg&w=16&q=75 1x, /_next/image?url=%2Ficon-plus.svg&w=32&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </button>
</div>
`;
